api/python/cellxgene_census/src/cellxgene_census/_open.py

```diff
@@ -151,7 +172,9 @@ def open_soma(
             "consistency."
         )
 
-    return _open_soma(description["soma"], context)
+    locator = _resolve_census_locator(description["soma"], selected_mirror)
+
+    return _open_soma(locator, context)
 
 
 def get_source_h5ad_uri(dataset_id: str, *, census_version: str = DEFAULT_CENSUS_VERSION) -> CensusLocator:
```

Comment:
The return type for this function is now wrong, or at least somewhat wrong.  It should return something more akin to the `ResolvedCensusLocator` contents, even if it isn't really locating a Census.  I'm not entirely sure what the right approach is here, given that we are soon moving to a model where the H5ADs are only available via HTTPS, and don't need a full locator (i.e., no knowledge of protocol, provider or region is required), and at that point, we could drastically simplify this to just return a naked URI....

Reply:
ps. if we just mark as deprecated until #89 is resolved, we can hold off any other changes including this one

Reply:
As per Pablo's update, I will leave this method as is and read the h5ads from the absolute URI.